commons/src/main/java/org/infinispan/commons/marshall/SerializeWith.java

```diff
@@ -33,4 +33,10 @@
     */
    Class<? extends Externalizer<?>> value();
 
+   /**
+    * TODO
+    * @return
+    */
+   SerializeValueMatcher valueMatcher() default SerializeValueMatcher.MATCH_ALWAYS;
```

Comment:
Doesn't seem like the best place for this, a user is much more likely to use `SerializeWith` for a a new value type than for a new lambda.